### PR TITLE
Add human notes to a referral

### PR DESCRIPTION
**Pull request by patinipat**
# Context

# Changes in this PR

Add a create assessment note form on the review and assess show page
Notes are displayed using the moj Timeline component

### Before
localhost_3000_review-and-assess_df0e4635-b71d-4ad2-8191-d74b1bcb27ad_df0e4635-b71d-4ad2-8191-d74b1bcb27ad_summary_success=false

### After
localhost_3000_review-and-assess_df0e4635-b71d-4ad2-8191-d74b1bcb27ad_df0e4635-b71d-4ad2-8191-d74b1bcb27ad_summary_success=true

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by bryangaledxw** [comment]
Giving this a preliminary pass, all looks good. Happy with the commit structure, building up the layers. I think maybe it'd be worth putting the new util method (and its inclusion in nunjucks setup) in it's own commit before the controller changes, but that's a minor thing. Some other things I spotted:

* Should `text: note.message` in assessment utils be `html: formatLines(note.message)`? Just want to make sure notes with line breaks are handled properly. Suggest it's worth using `faker.lorem.lines()` in the factory to exercise this
* Do you need any error handling in the new controller method. Does/should the API allow empty notes? You'd need to do the same style of error handling as in `PremisesController.create()` and add a new message to `errors.json`

**Review by libuk — COMMENTED** [review]
Looks good. I agree with @bryangaledxw that the utils change could be pulled out into a separate commit, but shouldn't be a blocker. I'll defer to him for approval as there may be some context I'm missing.

**Comment by patinipat** [comment]
Extract the new helper function in assessmentUtils in its own commit

File: server/views/temporary-accommodation/assessments/show.njk
Comment (bryangaledxw) [inline server/views/temporary-accommodation/assessments/show.njk]: Suggest use taTextarea here instead, it'll save some repetition. And use the same naming as the field in the API. If this is "message" rather than "notes", everything will link up by magic (ie. `catchValidationErrorOrPropogate`) if there's an error from the API - this'll be be highlighted in red, and the errors at the top of the page will link to here

File: server/utils/nunjucksSetup.ts
Comment (bryangaledxw) [inline server/utils/nunjucksSetup.ts]: So is this used internally by the timeline component?

File: cypress_shared/pages/assess/show.ts
Comment (bryangaledxw) [inline cypress_shared/pages/assess/show.ts]: Think this could be a call down to `completeTextInputByLabel` in the base class - `completeTextInputByLabel('Add a note', message)`. And getting more into personal preference, consider just doing something like `cy.get('button').contains('Save note').click()`

**Review by bryangaledxw — COMMENTED** [review]
Gone through a bit more closely. I'll hold off regarding as done until you've had a chat with Nick about showing the name of the person who created the note, but this doesn't look far off, good stuff

File: server/views/temporary-accommodation/assessments/show.njk
Comment (patinipat) [reply server/views/temporary-accommodation/assessments/show.njk]: :recycle:

File: cypress_shared/pages/assess/show.ts
Comment (patinipat) [reply cypress_shared/pages/assess/show.ts]: :recycle:

File: server/utils/nunjucksSetup.ts
Comment (patinipat) [reply server/utils/nunjucksSetup.ts]: yes https://github.com/ministryofjustice/moj-frontend/blob/main/src/moj/components/timeline/template.njk#L18

**Comment by patinipat** [comment]
Added support to display the name of the staff member who created the note. This relies on the `User` type being updated to provide the `name` property. This commit https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/pull/559/commits/678d2299e68e504bd32da39f36ebd13ac3bec44f is there in case we need to format the user.name has Im not sure what will format will be provided.

**Comment by bryangaledxw** [comment]
@libuk Ready for another look

File: wiremock/assessmentStubs.ts
Comment (libuk) [inline wiremock/assessmentStubs.ts:91]: What does this priority key do out of interest?

**Review by libuk — APPROVED** [review]
Nice work 👍🏾

File: wiremock/assessmentStubs.ts
Comment (bryangaledxw) [reply wiremock/assessmentStubs.ts:91]: Generally when we're stubbing a POST endpoint, we'll have the success response that matches against a path, and some error responses that match against the path plus specific fields being missing. We want to make sure that the success response takes lower priority, otherwise that'd match first and we'd never see the error response in our wiremock API

Worth having a look at `errorStub()` and `getCombinations()`. We don't use the latter here as there's only one field (`message`), but generally we use the two together to generate error stubs for any combination of missing fields

**Pull request merged**
